static/js/narrow.js

```diff
@@ -599,6 +599,46 @@ exports.by_recipient = function (target_id, opts) {
     }
 };
 
+// Called by the narrow_to_compose_target hotkey.
+exports.to_compose_target = function () {
+    if (!compose_state.composing()) {
+        return;
+    }
+
+    var opts = {
+        trigger: 'narrow_to_compose_target',
+    };
+
+    if (compose_state.get_message_type() === 'stream') {
+        var stream_name = compose_state.stream_name();
+        var stream_id = stream_data.get_stream_id(stream_name);
+        if (!stream_id) {
+            return;
+        }
+        var topics = topic_data.get_recent_names(stream_id);
+        var operators = [{operator: 'stream', operand: stream_name}];
+        var topic = compose_state.topic();
+        if (topics.indexOf(topic) !== -1) {
+            operators.push({operator: 'topic', operand: topic});
+        }
```

Comment:
What's the story with this block?  I would have thought we'd want to do this if and only if the topic in the compose box is nonempty, irrespective of `recent_topics` state.

Reply:
@rishig suggested that if the topic didn't exist, then we should just stay narrowed to the stream (but greyed out). I tried both approaches, and this feels a bit natural ij comparison to being dropped to an empty narrow.

Reply:
OK.  I guess we can try this and see, but this deserves a comment explaining why we're doing it.